orchagent/portsorch.cpp

```diff
@@ -529,142 +529,142 @@ void PortsOrch::doLagTask(Consumer &consumer)
     {
         KeyOpFieldsValuesTuple t = it->second;
 
-        string key = kfvKey(t);
-        size_t found = key.find(':');
-        string lag_alias, port_alias;
-        if (found == string::npos)
-            lag_alias = key;
+        string lag_alias = kfvKey(t);
+        string op = kfvOp(t);
+
+        if (op == SET_COMMAND)
+        {
+            /* Duplicate entry */
+            if (m_portList.find(lag_alias) != m_portList.end())
+            {
+                it = consumer.m_toSync.erase(it);
+                continue;
+            }
+
+            if (addLag(lag_alias))
+                it = consumer.m_toSync.erase(it);
+            else
+                it++;
+        }
+        else if (op == DEL_COMMAND)
+        {
+            Port lag;
+            /* Cannot locate LAG */
+            if (!getPort(lag_alias, lag))
+            {
+                it = consumer.m_toSync.erase(it);
+                continue;
+            }
+
+            if (removeLag(lag))
+                it = consumer.m_toSync.erase(it);
+            else
+                it++;
+        }
         else
         {
-            lag_alias = key.substr(0, found);
-            port_alias = key.substr(found+1);
+            SWSS_LOG_ERROR("Unknown operation type %s", op.c_str());
+            it = consumer.m_toSync.erase(it);
         }
+    }
+}
+
+void PortsOrch::doLagMemberTask(Consumer &consumer)
+{
+    if (!isInitDone())
+        return;
+
+    auto it = consumer.m_toSync.begin();
+    while (it != consumer.m_toSync.end())
+    {
+        KeyOpFieldsValuesTuple t = it->second;
```

Comment:
Not that light. Suggestion "auto& t".

---
In reply to: [111024651](https://github.com/Azure/sonic-swss/pull/187#discussion_r111024651) [](ancestors = 111024651)